Automattic/jetpack
# Refactor/wpcom token subscription service into jetpack token subscription service

Fixes 197-gh-automattic/gold

## Proposed changes:

* Merges the logic from `WPCOM_Token_Subscription_Service` into `Jetpack_Token_Subscription_Service`.
* Move `available` logic from `WPCOM_Token_Subscription_Service` to `is_wpcom` function.
* Make `available` logic in `Jetpack_Token_Subscription_Service` into `$this->is_wpcom() || <existing logic>`.
* Move WPCOM-specific functions behind `if ( $this->is_wpcom ) { return ... }` clause.
* Rename all instances of `WPCOM_Token_Subscription_Service` to `Jetpack_Token_Subscription_Service'.
* Delete `WPCOM_Token_Subscription_Service`.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Apply this patch to your sandbox and sandbox your site (unless using JN - in which case setup constants and global http), public api, and subscribe.
* If running on JN, activate this branch.
* Try all payment blocks work properly when logged in, not logged in and a new subscriber, not logged in but just subscribed, etc..
* Be sure to try this on WPCOM simple if you've tried on JN or vice-versa. 
* Make sure there are no regressions.
